edweiss-app/app/(app)/courses/[id]/index.tsx

```diff
@@ -87,54 +73,48 @@ const testIDs = {
  * 
  * @returns JSX.Element - The rendered component for the course page.
  */
-const CoursePage: React.FC<LightDarkProps> = ({ light, dark }) => {
-
-	// Colors
-	const backgroundColor = useThemeColor({ light, dark }, 'mantle');
-	const cherryColor = useThemeColor({ light, dark }, 'cherry');
-	const defaultColor = useThemeColor({ light, dark }, 'darkNight');
-	const urgentColor = useThemeColor({ light, dark }, 'yellow');
-	const headingsColor = useThemeColor({ light, dark }, 'darkBlue');
-	const swipeBackgroundColor1 = useThemeColor({ light, dark }, 'yellowlogo');
-	const swipeBackgroundColor2 = useThemeColor({ light, dark }, 'clearGreen');
-
-	// Define swipeableRefs
-	const swipeableRefs = useRef<(Swipeable | null)[]>([]);
+const CoursePage: ApplicationRoute = () => {
 
 	// Get course id from URL
 	const { id } = useLocalSearchParams();
 	if (typeof id !== 'string') return <Redirect href={'/'} />;
 
-	// Get course data from Firestore
+	// Get course & assignments data from Firestore
 	const [course] = usePrefetchedDynamicDoc(CollectionOf<Course>('courses'), id, undefined);
-	if (course == undefined) return <TActivityIndicator size={TActivityIndicatorSize} />;
+	const firebase_data = useDynamicDocs(CollectionOf<Assignment>(`courses/${id}/assignments`));
 
-	// console.log(`nb assignments: ${course.data.assignments.length}`);
-	// console.log(JSON.stringify(course));
+	if (course == undefined || firebase_data == undefined) return <TActivityIndicator size={40} />;
+
+	const assignmentsArray = firebase_data ? firebase_data.map(doc => ({ id: doc.id, data: doc.data })) : [];
 
 	// Sort assignments by due date and add color based on time difference
-	const assignments: AssignmentWithColor[] = course.data.assignments
-		.sort((a, b) => a.dueDate.seconds - b.dueDate.seconds) // Seconds comparison
+	const assignments: AssignmentWithColor[] = assignmentsArray
+		.sort((a, b) => a.data.dueDate.seconds - b.data.dueDate.seconds) // Seconds comparison
 		.map((assignment) => {
 			const currentTime = new Date().getTime(); // Actual time in millisecondes
-			const assignmentDueTime = assignment.dueDate.seconds * msInASecond; // Convert dueDate in millisecondes
+			const assignmentDueTime = assignment.data.dueDate.seconds * timeInMS.SECOND; // Convert dueDate in millisecondes
 			const timeDifference = assignmentDueTime - currentTime; // Difference between current time and due time
 
 			// Define color based on time difference
 			const color =
```

Comment:
To find a color, you can create a function instead of a variable to clear up the code.